homeassistant/components/shelly/coordinator.py

```diff
@@ -844,21 +847,14 @@ async def shutdown(self) -> None:
         """Shutdown the coordinator."""
         if self.device.connected:
             try:
-                if not self.sleep_period:
+                if not self.sleep_period and is_rpc_ble_scanner_supported(
+                    self.config_entry
+                ):
                     await async_stop_scanner(self.device)
                 await super().shutdown()
             except InvalidAuthError:
                 self.config_entry.async_start_reauth(self.hass)
                 return
-            except RpcCallError as err:
-                # Ignore 404 (No handler for) error
-                if err.code != 404:
-                    LOGGER.debug(
-                        "Error during shutdown for device %s: %s",
-                        self.name,
-                        err.message,
-                    )
-                return
```

Comment:
Instead of catching the error `is_rpc_ble_scanner_supported` is checking `rpc_supports_scripts` which is evaluated at startup by the same error, so if we have no scripts support we don't enable BLE and same for disable, instead of trying to disable and fail we can avoid disabling if we previously evaluated that the device doesn't support this.